builder/azure/common/template/template_builder.go

```diff
@@ -683,7 +699,7 @@ const BasicTemplate = `{
   },
   "variables": {
     "addressPrefix": "10.0.0.0/16",
-    "apiVersion": "2017-03-30",
+    "apiVersion": "2022-03-01",
```

Comment:
Does this PR require an API version bump this large to work?  If possible I would like to leave the API version the same and save API upgrading for a larger track of work we're planning here https://github.com/hashicorp/packer-plugin-azure/issues/243

Reply:
good question - I think I just updated to the latest version available but it does make sense to do this in a more general way. 

I'm not sure where to find the relevant information though. Looking at the [docs](https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/2020-12-01/virtualmachines?pivots=deployment-language-arm-template) it only shows until 2020-12-01 and not earlier. Not sure where I could check that. Do you know?

Reply:
Hrm I'm not sure 🤔 , could you manually test your PR without the API bump, see if it fails, and if it doesn't work maybe let's just do the minimum API bump required, usually the error message will say what the required API verison is, if it doesn't I would recommend trying 2020-12-01 as the furthest bump we do for now

Reply:
I found [ARM schema 2019-03-01](https://github.com/Azure/azure-resource-manager-schemas/blob/025cb02ce0c1730099b4c76857099142e2cbf025/schemas/2019-03-01/Microsoft.Compute.json) to be the earliest which has the necessary attributes.